posthog/models/event/query_event_list.py

```diff
@@ -145,7 +145,7 @@ def run_events_query(
 ) -> EventsQueryResponse:
     # Note: This code is inefficient and problematic, see https://github.com/PostHog/posthog/issues/13485 for details.
     # To isolate its impact from rest of the queries its queries are run on different nodes as part of "offline" workloads.
-    hogql_context = HogQLContext()
+    hogql_context = HogQLContext(legacy_person_property_handling=True)
```

Comment:
ok finally seeing this. unsure how much of a refactor it is, or if you're working on it in another PR/deprioritized this but if the issue is breaking things I'd just refactor and break it - it's too early to start branching for "legacy" syntax

Reply:
The idea is to still be able to use HogQL within existing property filters in existing insights, breakdowns, etc. I was also doubting if I should just remove this support or not, but decided to keep, as 1) it was quite OK to get it to work in the end (tests pass), 2) there are people on cloud using it, 3) if I remove it, then the alternative approach to hogql in property filters would only come after we rewrite all insights onto a hogql base, and who knows when that happens for everything (think paths, etc)

Reply:
Ah so `legacy_person_property_handling=True` means `is_running_in_insight_fllter_context` ?

Reply:
Yup, changed it to `within_non_hogql_query` as per your suggestion below